### ISSUE TITLE
Crash after sending checkmated FEN position

### ISSUE THREAD

**Issue by fohristiwhirl** [LeelaChessZero/lc0#888]
```
position fen rnb1kbnr/pppp1ppp/4p3/8/5PPq/8/PPPPP2P/RNBQKBNR w KQkq - 1 3 moves
go
```

This crashes the engine.

**Comment by fohristiwhirl** [comment]
Note that arriving there via a series of moves does not crash, e.g.

```
position startpos moves f2f4 e7e6 g2g4 d8h4
go
```

**Comment by mooskagh** [comment]
Probably fixed by #887.

**Comment by lealgo** [comment]
> Probably fixed by #887.

It does fix the fen of the first post.

@fohristiwhirl  Do you have another one to test?
